creature.cpp

```diff
@@ -1144,6 +1144,8 @@ static string getAttackParam(AttackType type) {
     case AttackType::SPELL: return "spell";
     case AttackType::POSSESS: return "touch";
   }
+  assert("__FUNCTION__ case unknown attack type");
+  return "";
```

Comment:
If you allow modding and/or move your data away from the hard coded approach, you'll need this type of last check.

As is, it is a debug tool as it'll catch edge case errors in your hardcoded definitions as there's no defaults.